backend/service/authn/authn.go

```diff
@@ -51,4 +52,5 @@ type Provider interface {
 	Verify(ctx context.Context, rawIDToken string) (*Claims, error)
 	GetAuthCodeURL(ctx context.Context, state string) (string, error)
 	Exchange(ctx context.Context, code string) (token *oauth2.Token, err error)
+	CreateToken(ctx context.Context, subject string, expiry time.Duration) (token *oauth2.Token, err error)
```

Comment:
this belongs in a separate Issuer interface

Reply:
I gave this a shot but it results in some awkward casts in some places, maybe I'm missing something?